fdw/pxf_bridge.c

```diff
@@ -192,11 +246,23 @@ PxfBridgeImportStart(PxfFdwScanState *pxfsstate)
 					 pxfsstate->retrieved_attrs,
 					 pxfsstate->projectionInfo);
 
-	pxfsstate->churl_handle = churl_init_download(pxfsstate->uri.data, pxfsstate->churl_headers);
+	if (strcmp("https", pxfsstate->options->pxf_protocol) == 0) {
+		ssl_options = churl_make_ssl_options(pxfsstate->options);
+	}
+
+	pxfsstate->churl_handle = churl_init_download(pxfsstate->uri.data, pxfsstate->churl_headers, ssl_options);
+	if (ssl_options != NULL) {
+		free_churl_ssl_options(ssl_options);
+	}
```

Comment:
why not simply
```suggestion
	pxfsstate->churl_handle = churl_init_download(pxfsstate->uri.data, pxfsstate->churl_headers);
	if (strcmp("https", pxfsstate->options->pxf_protocol) == 0) {
		churl_set_ssl_options(pxfsstate->options);
	}
```
and in external table do similar thing

Reply:
ssl_options has a type churl_ssl_options which has defined in libchurl, while pxfsstate->options is of type PxfOptions defined in the Pxf. Idea of making churl_ssl_options was to not introduce into libchurl extra dependency of higher level PXF.

Reply:
The `churl_set_ssl_options` function could be called differently and placed outside of `libchurl`.

Reply:
SSL logically looks very like an connection properties and I decided to set them as a part of libchurl.

Reply:
Why not do something similar to `pxfsstate->churl_headers` with a common type for `SSL` options?

Reply:
pxfsstate->churl_headers is a void pointer and does not discover specific type structure. Unlike it, churl_ssl_options is typed. Placing it to the pxfsstate breaks the fact that pxfsstate does not now discover churl-dependent things.
Also, instances of churl_ssl_options are used locally and independently, I don't see any need to put it into pxfsstate.